src/armTemplates/resources/appInsights.ts

```diff
@@ -0,0 +1,46 @@
+import { ArmResourceTemplateGenerator } from "../../models/armTemplates";
+import { ServerlessAzureConfig, ResourceConfig } from "../../models/serverless";
+
+export const AppInsightsResource: ArmResourceTemplateGenerator = {
+  getTemplate: () => {
+    return {
+      "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+      "contentVersion": "1.0.0.0",
+      "parameters": {
+        "appInsightsName": {
+          "defaultValue": "",
+          "type": "String"
+        },
+        "location": {
+          "defaultValue": "",
+          "type": "String"
+        }
+      },
+      "variables": {},
+      "resources": [
+        {
+          "apiVersion": "2015-05-01",
+          "name": "[parameters('appInsightsName')]",
+          "type": "microsoft.insights/components",
+          "location": "[parameters('location')]",
+          "properties": {
+            "Application_Type": "web",
+            "ApplicationId": "[parameters('appInsightsName')]",
+            "Request_Source": "IbizaWebAppExtensionCreate"
+          }
+        }
+      ]
+    }
+  },
+
+  getParameters: (config: ServerlessAzureConfig) => {
+    const resourceConfig: ResourceConfig = {
+      name: `${config.provider.prefix}-${config.provider.region}-${config.provider.stage}-appinsights`,
```

Comment:
Maybe a function that takes in a config and spits out this naming convention? I think I see it duplicated a few times. Plus, I'd like to leverage the same naming convention with the blob storage work

Reply:
Agreed - will see what i can do.